### ISSUE TITLE
Capitalized api keys values convert to simple

### ISSUE THREAD

**Issue by myas5868** [microsoft/playwright#21492]
### Discussed in https://github.com/microsoft/playwright/discussions/21369

<div type='discussions-op-text'>

<sup>Originally posted by **myas5868** March  3, 2023</sup>
I'm sending a post request 

`const response = await request.post(path,{
            headers: {
                'Accept': apiHeaders().ACCEPT,
                'Content-Type': apiHeaders().CONTENT_TYPE,
                'X-OO-SERVLET-TOKEN': apiHeaders().X_OO_SERVLET_TOKEN
            },
            data: JSON.stringify(orderData().TOMORROWDELIVERYDATE)
        });`

but X-OO-SERVLET-TOKEN is converted to  simple case in header like below:

POST [https:/localhost/api/order]
accept: application/json
accept-encoding: gzip,deflate,br
content-type: application/json
x-oo-servlet-token: 680242a2a4d811a8a54eb6c6f33a
content-length: 444

expected header should be

POST [https:/localhost/api/order]
accept: application/json
accept-encoding: gzip,deflate,br
content-type: application/json
X-OO-SERVLET-TOKEN: 680242a2a4d811a8a54eb6c6f33a
content-length: 444</div>

**Comment by mxschmitt** [comment]
According to the HTTP/1.1 specification (Section 4.2 of the HTTP/1.1 specification (RFC 7230)), HTTP header field names are case-insensitive. This means that the field names can be written in any combination of upper or lower case letters and still be considered the same field.

Does your backend not like it when the `X-OO-SERVLET-TOKEN` is lowercase?

**Comment by myas5868** [comment]
yes, your understanding is correct. if X-OO-SERVLET-TOKEN is in lowercase it throws a 403 error.

<img width="720" alt="Screenshot 2023-03-09 at 10 22 15 AM" src="https://user-images.githubusercontent.com/43173414/223921510-cc865f0e-bcf7-439d-872f-09aebaad4745.png">

**Comment by myas5868** [comment]
and is there any way we can send the request with capitalized header key in the post request?
